bittide/src/Clash/Signal/TH/Extra.hs

```diff
@@ -0,0 +1,59 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+module Clash.Signal.TH.Extra where
+
+import Prelude
+
+import Clash.Signal
+import Data.String.Interpolate
+import GHC.Records (HasField (..))
+import Language.Haskell.TH
+
+{- | Derive instances of 'HasField' for a Signals of Records:
+
+Example:
+
+@
+data MyRecord n = MyRecord { field1 :: BitVector n, field2 :: BitVector n (Unsigned 8) }
+@
```

Comment:
```suggestion
> data MyRecord n = MyRecord { field1 :: BitVector n, field2 :: BitVector n (Unsigned 8) }
```

Reply:
`@` leaves a lot of "special" syntax, which messes up code blocks in Haddock. `>` doesn't accept any special syntax.